From 7801d429fa12ba5ff7453acd1085f0f9f8b13863 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 27 Jun 2022 17:59:49 +0200
Subject: Wayland: do not use GN wayland includes if use_system_libwayland is
 set

Build was failing in several components if use_system_libwayland was
enabled, as it would still include third_party/wayland headers. Those
could be clashing with the system wayland headers.

On top of that, as we depend on wayland-egl-backend.h for Vulkan
wayland support, and that is not a dependency checked in Chromium,
this adds a explicit dependency.

Bug: chromium:1340181
Change-Id: Ia7dff3b3c6ca034ab55113cb9337050bc7f66a51
---
 BUILD.gn                              | 14 ++++++++++----
 src/common/vulkan/BUILD.gn            |  2 +-
 src/libANGLE/renderer/vulkan/BUILD.gn |  5 ++++-
 src/third_party/volk/BUILD.gn         |  2 +-
 4 files changed, 16 insertions(+), 7 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 3d94badcd..830a62449 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1052,12 +1052,18 @@ if (angle_use_wayland) {
         "wayland-client",
         "wayland-egl",
       ]
+    } else {
+      include_dirs = [
+        "$wayland_dir/egl",
+        "$wayland_dir/src",
+      ]
     }
+  }
 
-    include_dirs = [
-      "$wayland_dir/egl",
-      "$wayland_dir/src",
-    ]
+  if (use_system_libwayland) {
+    pkg_config("angle_wayland_egl_backend_config") {
+      packages = [ "wayland-egl-backend" ]
+    }
   }
 
   group("angle_wayland") {
diff --git a/src/common/vulkan/BUILD.gn b/src/common/vulkan/BUILD.gn
index 9cb8fc30c..5933d001c 100644
--- a/src/common/vulkan/BUILD.gn
+++ b/src/common/vulkan/BUILD.gn
@@ -33,7 +33,7 @@ if (angle_enable_vulkan || angle_build_vulkan_system_info) {
     if (angle_shared_libvulkan) {
       defines = [ "ANGLE_SHARED_LIBVULKAN=1" ]
     }
-    if (angle_use_wayland) {
+    if (angle_use_wayland && !use_system_libwayland) {
       include_dirs = [ "$wayland_dir/src" ]
     }
     configs = [ "$angle_root:angle_vulkan_wayland_config" ]
diff --git a/src/libANGLE/renderer/vulkan/BUILD.gn b/src/libANGLE/renderer/vulkan/BUILD.gn
index 435fb0976..05a7cacd3 100644
--- a/src/libANGLE/renderer/vulkan/BUILD.gn
+++ b/src/libANGLE/renderer/vulkan/BUILD.gn
@@ -311,7 +311,10 @@ angle_source_set("angle_vulkan_backend") {
   public_configs = [ ":angle_vulkan_backend_config" ]
 
   if (angle_use_wayland) {
-    public_configs += [ "$angle_root:angle_wayland_config" ]
+    public_configs += [
+      "$angle_root:angle_wayland_config",
+      "$angle_root:angle_wayland_egl_backend_config",
+    ]
   }
 
   data_deps = []
diff --git a/src/third_party/volk/BUILD.gn b/src/third_party/volk/BUILD.gn
index 576f5c4f0..d827ee068 100644
--- a/src/third_party/volk/BUILD.gn
+++ b/src/third_party/volk/BUILD.gn
@@ -20,7 +20,7 @@ source_set("volk") {
   public_configs = [ ":volk_config" ]
   configs += [ "$angle_root:angle_no_cfi_icall" ]
   public_deps = [ "$angle_vulkan_headers_dir:vulkan_headers" ]
-  if (angle_use_wayland) {
+  if (angle_use_wayland && !use_system_libwayland) {
     include_dirs = [ "$wayland_dir/src" ]
   }
 }
-- 
2.34.1

